Client/mods/deathmatch/logic/luadefs/CLuaVectorGraphicDefs.cpp

```diff
@@ -0,0 +1,318 @@
+                {
+                    if (!pFile->Load(pParentResource, CScriptFile::MODE_READ))
+                    {
+                        delete pVectorGraphic;
+                        delete pFile;
+
+                        m_pScriptDebugging->LogCustom(luaVM, SString("Unable to load SVG (file doesn't exist) [%s]", pathOrRawData.value().c_str()));
+                        return false;
+                    }
+                    else
+                    {
+                        SString strXmlData;
+                        pFile->Read(pFile->GetSize(), strXmlData);
+
+                        // We don't need the file handler anymore
+                        delete pFile;
+
+                        if (strXmlData.empty() || !pVectorGraphic->LoadFromString(strXmlData))
+                        {
+                            delete pVectorGraphic;
+
+                            m_pScriptDebugging->LogCustom(luaVM, "Unable to load SVG file (check for XML syntax errors)");
+                            return false;
+                        }
+                    }
+                }
+            }
+            else
+            {
+                delete pVectorGraphic;
+
+                m_pScriptDebugging->LogCustom(luaVM, SString("Unable to load SVG (invalid file path) [%s]", pathOrRawData.value().c_str()));
+                return false;
+            }
+        }
+    }
+
+    // Make our element a child of the resource's file root
+    pVectorGraphic->SetParent(pParentResource->GetResourceDynamicEntity());
+
+    // Set our owner resource
+    pVectorGraphic->SetResource(pParentResource);
+
+    return pVectorGraphic;
+}
+
+CXMLNode* CLuaVectorGraphicDefs::SVGGetDocumentXML(CClientVectorGraphic* pVectorGraphic)
+{
+    return pVectorGraphic->GetXMLDocument();
+}
+
+bool CLuaVectorGraphicDefs::SVGSetDocumentXML(CClientVectorGraphic* pVectorGraphic, CXMLNode* pXMLNode, std::optional<CLuaFunctionRef> luaFunctionRef)
+{
+    if (luaFunctionRef.has_value() && VERIFY_FUNCTION(luaFunctionRef.value()))
+    {
+        CLuaFunctionRef funcRef = luaFunctionRef.value();
+
+        CLuaShared::GetAsyncTaskScheduler()->PushTask<bool>(
+            [funcRef, pVectorGraphic, pXMLNode] {
+                lua_State* luaVM = funcRef.GetLuaVM();
+                CLuaMain*  pLuaMain = m_pLuaManager->GetVirtualMachine(luaVM);
+
+                if (!pLuaMain || !pVectorGraphic)
+                    return false;
+
+                return pVectorGraphic->SetDocument(pXMLNode);
+            },
+            [funcRef](const bool didLoad) {
+                CLuaMain* pLuaMain = m_pLuaManager->GetVirtualMachine(funcRef.GetLuaVM());
+                if (pLuaMain)
+                {
+                    CLuaArguments arguments;
+                    arguments.PushBoolean(didLoad);
+                    arguments.Call(pLuaMain, funcRef);
+                }
+            });
+
+        return true;
+    }
+
+    return pVectorGraphic->SetDocument(pXMLNode);
```

Comment:
The code is currently duplicated twice, once for the callback and once for the non-callback. I would like this to be refactored into a e.g. single private function call.